ipwb/reconstructive.js

```diff
@@ -1,116 +1,145 @@
-    if(opts instanceof Object) {
-      Object.assign(config, opts);
-      derivedConfig();
-      config.debug && console.log(`${NAME}:${VERSION} initialized with supplied configs`);
-    } else {
-      console.warn('Expected an object not a', typeof opts);
+    /**
+     * Identifier of the module, sent to the server as X-ServiceWorker header.
+     * Defaults to the name and version of the module.
+     *
+     * @type {string}
+     */
+    this.id = `${this.NAME}:${this.VERSION}`;
+
+    /**
+     * The format of URI-Ms (e.g., http://example.com/archive/<datetime>/<urir>).
+     *
+     * @type {string}
+     */
+    this.urimPattern = `${self.location.origin}/memento/<datetime>/<urir>`;
+
+    /**
+     * The URL or absolute path of the JS file that defines custom banner element.
+     * Only necessary if showBanner is set to true.
+     *
+     * @type {string}
+     */
+    this.bannerElementLocation = `${self.location.origin}/reconstructive-banner.js`;
+
+    /**
+     * Whether or not to show an archival banner.
+     * Defaults to false.
+     *
+     * @type {boolean}
+     */
+    this.showBanner = false;
+
+    /**
+     * Whether or not to show debug messages in the console.
+     * Defaults to false.
+     *
+     * @type {boolean}
+     */
+    this.debug = false;
+
+    // Iterate over the supplied configuration object to overwrite defaults and add new properties
+    if (config instanceof Object) {
+      for (const [k, v] of Object.entries(config)) {
+        /** @ignore **/
+        this[k] = v;
+      }
     }
-  }
 
- /**
-  * exclusions - An object of functions to check whether the request should be excluded from being rerouted.
-  *              Add more members to the object to add more exclusions or modify/delete existing ones.
-  *              The property name can be anything descriptive of the particular exclusion, which will be shown in debug logs.
-  *              Each member function is called with the fetch event and config object as parameters.
-  *              If any member returns true, the fetch event is excluded from being rerouted.
-  *
-  * @public
-  */
-  let exclusions = {
-    notGet: (event, config) => event.request.method != 'GET',
-    bannerElement: (event, config) => config.showBanner && event.request.url.endsWith(config.bannerElementLocation),
-    localResource: (event, config) => !(config.urimRegex.test(event.request.url) || config.urimRegex.test(event.request.referrer))
-  };
+    /**
+     * A private object with varius RegExp properties (possibly derived from other properties) for internal use.
+     * 
+     * @private
+     * @type    {{urimPattern: RegExp, absoluteRefrence: RegExp, bodyEnd: RegExp}}
+     */
+    this._regexps = {
+      urimPattern: new RegExp(`^${this.urimPattern.replace('<datetime>', '(\\d{14})').replace('<urir>', '(.*)')}$`),
+      absoluteRefrence: new RegExp(`(<(iframe|a).*?\\s+(src|href)\\s*=\\s*["']?)(https?:\/\/[^'"\\s]+)(.*?>)`, 'ig'),
```

Comment:
Should this be` absoluteReference`? It is consistently misspelled and should still work regardless.

Reply:
That's a typo and should be fixed in the upstream as well. Thanks for pointing this out.

Reply:
Experimented with the `Co-Authored-By` for the first time. :-)